src/wire/libevent_server.cpp

```diff
@@ -112,17 +115,48 @@ void LibeventServer::StartServer() {
       throw ConnectionException("Failed to create listen socket");
     }
 
+    int conn_backlog = 12;
     int reuse = 1;
     setsockopt(listen_fd, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(reuse));
 
-    if (bind(listen_fd, (struct sockaddr *)&sin, sizeof(sin)) < 0) {
-      throw ConnectionException("Failed to bind socket to port: " +
-                                std::to_string(port_));
+    /* Initialize SSL listener connection */
+    SSL_load_error_strings();
+    SSL_library_init();
+
+    if ((ssl_context = SSL_CTX_new(TLSv1_server_method())) == NULL)
+    {
+      throw ConnectionException("Error creating SSL context.\n");
     }
 
-    int conn_backlog = 12;
-    if (listen(listen_fd, conn_backlog) < 0) {
-      throw ConnectionException("Failed to listen to socket");
+    LOG_INFO("private key file path %s", private_key_file_.c_str());
+    /* register private key */
+    if (SSL_CTX_use_PrivateKey_file(ssl_context, private_key_file_.c_str(),
```

Comment:
Will the code be more concise if you use try-catch blocks to implement the logic from Row 133 to Row 159?